### PR TITLE
10666 Add button to configure Related resource relationship type

### PR DESCRIPTION
**Pull request by whatisgalen**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Issues Solved

#10666

### Checklist

-   [x] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: @scholiumtech
*   Found by: @whatisgalen 
*   Tested by: @jacobtylerwalls 
*   Designed by: @whatisgalen

### CODE REVIEW

File: arches/app/templates/views/components/datatypes/resource-instance.htm [hunk 1]
```diff
@@ -35,13 +35,19 @@
...
-                        data-bind="
+                    <div class='rr-table-column icon-column'>
+                        <button 
+                            data-bind="
                             click: function() { if($parent.isEditable !== false){self.toggleSelectedResource($data)}; },
                             clickBubble: false,
                             onEnterkeyClick, onSpaceClick," 
-                        style="cursor: pointer;">
+                            style="cursor: pointer;">
+                            <i class="fa fa-gear"></i>
```
Comment (jacobtylerwalls) [inline arches/app/templates/views/components/datatypes/resource-instance.htm:45]: This has meaning, so let's add an `aria-label`. Maybe there is a suitable string already in javascript.htm like `edit`?

File: arches/app/templates/views/resource/related-resources/related-resources-manager.htm [hunk 2]
```diff
@@ -138,7 +138,12 @@ <h3 class="">
                                     <i class="fa fa-info-circle"></i>
                                 </button>
                             </div>
-                            <div class="rr-table-column" style="flex-grow: 1;" data-bind="click: self.toggleSelectedResourceRelationship, clickBubble: false">
+                            <div class='rr-table-column icon-column'>
+                                <button data-bind="click: self.toggleSelectedResourceRelationship, clickBubble: false">
+                                    <i class="fa fa-gear"></i>
```
Comment (jacobtylerwalls) [inline arches/app/templates/views/resource/related-resources/related-resources-manager.htm:143]: same

File: arches/app/templates/views/components/datatypes/resource-instance.htm [hunk 3]
```diff
@@ -35,13 +35,19 @@
...
                         </button>
                     </div>
-                    <div class="rr-table-column" tabindex="0" role="button" style="flex-grow: 1; padding-left: 10px;"
-                        data-bind="
+                    <div class='rr-table-column icon-column'>
+                        <button 
+                            data-bind="
                             click: function() { if($parent.isEditable !== false){self.toggleSelectedResource($data)}; },
                             clickBubble: false,
                             onEnterkeyClick, onSpaceClick,"
```
Comment (jacobtylerwalls) [inline arches/app/templates/views/components/datatypes/resource-instance.htm:43]: With this change, the space action no longer works; it just flickers:


https://github.com/archesproject/arches/assets/38668450/da170fad-a622-4867-983e-fd648fa3e9d7

File: arches/app/templates/views/components/datatypes/resource-instance.htm [hunk 4]
```diff
@@ -35,13 +35,19 @@
...
+                    <div class='rr-table-column icon-column'>
+                        <button 
+                            data-bind="
                             click: function() { if($parent.isEditable !== false){self.toggleSelectedResource($data)}; },
                             clickBubble: false,
                             onEnterkeyClick, onSpaceClick," 
-                        style="cursor: pointer;">
+                            style="cursor: pointer;">
+                            <i class="fa fa-gear"></i>
+                        </button>
```
Comment (jacobtylerwalls) [inline arches/app/templates/views/components/datatypes/resource-instance.htm:46]: When the node is not editable, we should style the icon as disabled, just like delete:

disabled-style

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]

**Comment by chiatt** [comment]
@whatisgalen, Could you rename this PR to something more descriptive, e.g. `Button for resource instance relationship type config`?

File: arches/app/templates/javascript.htm [hunk 5]
```diff
@@ -189,6 +189,7 @@
         select-a-nodegroup='{% trans "Select a nodegroup" as selectANodegroup %} "{{ selectANodegroup|escapejs }}"'
         no-relationships-added='{% trans "No Relationships Added" as noRelationshipsAdded %} "{{ noRelationshipsAdded|escapejs }}"'
         relate-resource='{% trans "Relate Resource" as relateResource %} "{{ relateResource|escapejs }}"'
+        configure-related-resource-relationship='{% trans "Configure Related Resource Relationship" as configureRelateResourceRelationship %} "{{ configureRelateResourceRelationship|escapejs }}"'
```
Comment (jacobtylerwalls) [inline arches/app/templates/javascript.htm:192]: Just a heads up: the left side is `related` and the right side is `relate`: would be great to have both the same

**Comment by whatisgalen** [comment]
I think there are some other bugs related to this UI. When I tried testing as of the latest merge from `dev/7.6.x`, the first issue is that this check in the resource-instance widget template  `<!-- ko if: self.relationship() -->` was returning false (undefined) for no apparent reason. Both resource models had ontologies set up and are relatable to each other. (This persisted when I switched from the feature branch to `dev/7.6.x`) When I removed that check from the htm file, the valueid for the relationship type appeared instead of the preflabel.

Screenshot 2024-03-18 at 4 33 41 PM

File: arches/app/templates/javascript.htm (hunk 5 above)
Comment (whatisgalen) [reply arches/app/templates/javascript.htm:192]: Whoops that is definitely a typo

File: arches/app/templates/views/components/datatypes/resource-instance.htm (hunk 1 above)
Comment (whatisgalen) [reply arches/app/templates/views/components/datatypes/resource-instance.htm:47]: requested changes in latest

File: arches/app/templates/views/components/datatypes/resource-instance.htm (hunk 3 above)
Comment (jacobtylerwalls) [reply arches/app/templates/views/components/datatypes/resource-instance.htm:45]: Still seeing this in the latest, did you have a look?

I'm showing this is fixed with
```diff
diff --git a/arches/app/templates/views/components/datatypes/resource-instance.htm b/arches/app/templates/views/components/datatypes/resource-instance.htm
index 60d6eb9f7..eceb1ff07 100644
--- a/arches/app/templates/views/components/datatypes/resource-instance.htm
+++ b/arches/app/templates/views/components/datatypes/resource-instance.htm
@@ -41,8 +41,7 @@
                             data-bind="
                             click: function() { if($parent.isEditable !== false){self.toggleSelectedResource($data)}; },
                             attr: {'aria-label': $root.translations.configureRelatedResourceRelationship, 'disabled': $parent.isEditable === false,},
-                            clickBubble: false,
-                            onEnterkeyClick, onSpaceClick," 
+                            clickBubble: false,"
                             style="cursor: pointer;">
                             <i class="fa fa-gear"></i>
                         </button>
```

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
Thanks for updates, just two little nits and we can get this in 👍

**Comment by jacobtylerwalls** [comment]
Couple more things just occurred to me.

- Would be nice to remove the cursor pointer and tabindex from the resource label if it's no longer interactive
- Did you consider making these changes in resource-instance-creator.htm also?

File: arches/app/templates/views/components/datatypes/resource-instance.htm (hunk 4 above)
Comment (whatisgalen) [reply arches/app/templates/views/components/datatypes/resource-instance.htm:48]: fix in latest

File: arches/app/templates/views/components/datatypes/resource-instance.htm (hunk 3 above)
Comment (whatisgalen) [reply arches/app/templates/views/components/datatypes/resource-instance.htm]: fix in latest

**Comment by whatisgalen** [comment]
> Couple more things just occurred to me.
> 
> * Would be nice to remove the cursor pointer and tabindex from the resource label if it's no longer interactive ✅ 
> * Did you consider making these changes in resource-instance-creator.htm also?

Is there a separate UX for creating resource-instance datatype data in `related-instance-creator.js`? I'm seeing that it just uses the existing widgets.


https://github.com/archesproject/arches/assets/8366985/585ee71e-a861-410c-9fff-19c20dfc1a77

**Review by jacobtylerwalls — APPROVED** [review]
+1 tested. Thanks for the updates!

**Pull request merged**